internal/controller/aws_interfaces.go

```diff
@@ -7,9 +7,16 @@ import (
 
 type IamWrapper interface {
 	GetUser(userName string) (*types.User, error)
+	IsExistingUser(userName string) (bool, error)
+	HasLoginProfile(userName string) (bool, error)
+	HasAccessKey(userName string) (bool, error)
+	ListGroupsForUser(userName string) ([]types.Group, error)
 	CreateUser(userName string) (*types.User, error)
+	CreateUserIfNotExists(userName string) error
 	ListUsers(maxUsers int32) ([]types.User, error)
 	CreateLoginProfile(password string, userName string, passwordResetRequired bool) (types.LoginProfile, error)
+	CreateLoginProfileIfNotExists(password string, userName string, passwordResetRequired bool) error
 	CreateAccessKeyPair(userName string) (*types.AccessKey, error)
 	AddUserToGroup(groupName string, userName string) (middleware.Metadata, error)
+	RemoveUserFromGroup(groupName string, userName string) (middleware.Metadata, error)
```

Comment:
I see a lot of usage of `context.TODO()` in the implementation of these functions. I'd update the signature of the functions to receive the context and pass it to the AWS API when needed. This allows to support context cancellation and if needed, things like timeouts as we were mentioning in the call